### ISSUE TITLE
🐛 [Bug Report] WotLK - Passenger not boarding vehicles (Tundra Mamoth, Motorcycle) correctly.

### ISSUE THREAD

**Issue by kelbren** [cmangos/issues#3663]
### Bug Details

Player has mount that can allow passengers, and passenger on another account tries to board, the player is still standing next to the vehicle, however the passenger is actually boarded from the point of view of the other account.
If passenger ejects from vehicle and does not move, it then shows the passenger actually getting onboard.
Confusing? yes. 

Player A has motorcycle, 
Player B tries to board  motorcycle.
Player B appears everything is correct.
Player A sees player B attempt to board (player B jumps slightly), but is now just standing frozen and not boarded.
if Player B ejects, and does not move, player A sees Player B board properly, at this point if player B moves, Player A sees player B removed from vehicle.

If I recall correctly, this used to work, I am not sure when this odd behavior started.

### Steps to Reproduce

1. Have another player attempt to mount a vehicle
2.  
3.
4.

### Expected behavior

Normal boarding and un-boarding of vehicles.

### Core SHA1 Commit Hash

ce715cae771a5950c743ef6067d914f3b232bd4f

### Database SHA1 Commit Hash

49c9b2ff84bc79248af9f3a7dd62f194f4223623

### Operating System

Ubuntu 20.04

### Client Version

3.3.5a (Wrath of the Lich King)

**Comment by insunaa** [comment]
This has been addressed in my Flame Leviathan PR iirc. it's just not merged yet
Edit: nvm it seems like I broke it again. I'll fix in a bit

**Comment by insunaa** [comment]
Yeah problem is this here:
![image](https://github.com/cmangos/issues/assets/3718129/dfad54d5-20c6-49b5-b071-2526fc59b9e6)
Get rid of that immobilization and all works as expected.
It'll be upstreamed once my Leviathan PR is merge-ready

**Comment by kelbren** [comment]
I commented out that line and it does appear to solve the issue for now.
Thanks for all your amazing work it is very much appreciated.

**Comment by al3xc1985** [comment]
@insunaa isn't your leviathan merge ready?

**Comment by insunaa** [comment]
No, because nobody is testing it. I'll give it a few more days, then I'll rebase and just merge it in its incomplete state.

**Comment by al3xc1985** [comment]
I can;t, I would have tested it, but just can;t alone. Maybe when bots for wotlk will go out, maybe I will give it a try with them

**Comment by kelbren** [comment]
completed

**Comment by Kashemir001** [comment]
@kelbren actually not yet, still in draft state https://github.com/cmangos/mangos-wotlk/pull/429
